sim/core/attack.go

```diff
@@ -643,6 +643,26 @@ func (aa *AutoAttacks) DelayMeleeBy(sim *Simulation, delay time.Duration) {
 	}
 }
 
+// PauseMeleeBy will prevent any swing from completing for the specified time.
+// This replicates a /stopattack and /startattack with a brief "pause" in the middle.
+// It's possible that no swing time is lost if the pauseTime is less than the remaining swing time.
+// Used by Rogue Gouge
+func (aa *AutoAttacks) PauseMeleeBy(sim *Simulation, pauseTime time.Duration) {
```

Comment:
What's the difference to the existing `StopMeleeUntil` ?

Reply:
`StopMeleeUntil` always changes the time the weapon should swing at (this effectively stops the swing timer from counting down); `PauseMeleeBy` only updates the `swingAt` if it would've occurred during that pause.

I manually confirmed this is how Gouge should operate, so it needed this specific functionality.